### PR TITLE
Torch model save and inference enable

### PR DESCRIPTION
**Pull request by wangzm509**
This code can sava model as torch and can use it to inference.
Update requirements.txt suit for Py39 and Py3.10.

### CODE REVIEW

File: ACPO-model/src/settings.py
Comment (wangzm509) [inline ACPO-model/src/settings.py:1]: the type for model saving

File: ACPO-model/src/train.py [hunk 1]
```diff
@@ -328,10 +328,16 @@ def learn(args, train_data_path:str, test_data_path:str, log_dir:str, train_kwar
...
                 if args.save_model:
-                    torch.save(model.state_dict(), os.path.join(log_dir, "plu.pt"))
-                    # bypass save_bp
-                    # save_pb(model, log_dir, num_features)
-                    save_pb_tmp_fuc(model, log_dir, num_features)
+                    print("Start saving models")
+                    if args.save_model_type == 'torch':
+                        # save model
+                        scrip_model = torch.jit.script(model)
+                        scrip_model.save('./log/inline_modelV0.0/modelfi.pt')
```
Comment (wangzm509) [inline ACPO-model/src/train.py:335]: add param ctrl

**Comment by yinrun** [comment]
squash all the commit together.

File: MLInference.py [hunk 2]
```diff
@@ -56,10 +57,22 @@ def ACPO_LOG(msg):
...
 
-def get_imported_and_infer(model_dir, signature):
+
+def get_torch_imported_and_infer(model_dir, signature):
+  """
+  Return the imported model and using the
+  given model_dir
+  """
+  model_path = os.path.join(os.path.dirname(__file__), model_dir)
+  torch_model_name = "modelfi.pt"
```
Comment (tsczajkowski) [inline MLInference.py:70]: Why is this hardcoded? I am pretty sure that the name of the model can be stored in the ACPO file and accessed here appropriately. Lets make this properly data-driven.

File: MLInference.py [hunk 3]
```diff
@@ -129,9 +142,20 @@ def load_model(model_spec_file):
   model_info_dict['outputlist'] = output_list
   model_dir = model_info_dict.get('modeldirectory')
   signature = model_info_dict.get('signature')
-  imported_and_infer = get_imported_and_infer(model_dir, signature)
-  imported = imported_and_infer[0]
-  infer = imported_and_infer[1]
+
+  load_model_type = model_info_dict.get('loadmodeltype')
```
Comment (tsczajkowski) [inline MLInference.py:146]: So I think using labels like loadmodeltype is fine. But I am seeing we are starting to have a lot of those labels. It is very easy for someone to mistype the string. Can we store these keys in a table and access them using something like predefine macros?

**Review by tsczajkowski — CHANGES_REQUESTED** [review]
Please fix highlighted items.

File: MLInference.py (hunk 3 above)
Comment (wangzm509) [reply MLInference.py]: done, make all "key" used by macros

File: ACPO-model/src/train.py [hunk 4]
```diff
@@ -328,10 +328,14 @@ def learn(args, train_data_path:str, test_data_path:str, log_dir:str, train_kwar
...
 
                 if args.save_model:
-                    torch.save(model.state_dict(), os.path.join(log_dir, "plu.pt"))
-                    # bypass save_bp
-                    # save_pb(model, log_dir, num_features)
-                    save_pb_tmp_fuc(model, log_dir, num_features)
+                    print("Start saving models")
+                    if args.save_model_type == 'torch':
+                        scrip_model = torch.jit.script(model)
+                        scrip_model.save(os.path.join(log_dir, "modelfi.pt"))
```
Comment (tsczajkowski) [inline ACPO-model/src/train.py:334]: The file names here should not be hardcoded. Please fix.

**Review by tsczajkowski — CHANGES_REQUESTED** [review]
Missed one location where filenames are hardcoded. Please fix

File: ACPO-model/src/train.py (hunk 4 above)
Comment (wangzm509) [reply ACPO-model/src/train.py]: Thanks！Done.

**Review by tsczajkowski — APPROVED** [review]
